### DISCUSSION TITLE
Virtualbox install on Elementary OS 8.x

### DISCUSSION THREAD

**Discussion by Shaggy1444** [elementary/triage#672] [Q&A]
Disclaimer: use at own risk; no support of the below is provided; not responsible for any issues

Most of this install will be done using the CLI/Terminal, so open it and switch to the Downloads folder (cd Downloads).
Read the instructions below first to become familiar with what needs to be done.

1.  add the following to /etc/apt/sources.list
        deb [arch=amd64 signed-by=/usr/share/keyrings/oracle-virtualbox-2016.gpg]  https://download.virtualbox.org/virtualbox/debian jammy contrib
        -make sure that jammy is the release being targeted
2.  download and register the Oracle public key
        wget -O- https://www.virtualbox.org/download/oracle_vbox_2016.asc | sudo gpg --yes --output /usr/share/keyrings/oracle-virtualbox-2016.gpg --dearmor
3.  now install virtualbox 7.0
        sudo apt-get update
        sudo apt install -y linux-headers-$(uname -r) dkms
        sudo apt-get install virtualbox-7.0
        -ensure the password you typed for the MOK enrollment is written down somewhere
4.  if a message indicates that libvpx7 cannot be installed, do this step and step 5
        -switch to your Downloads directory using the CLI/Terminal
        wget https://archive.ubuntu.com/ubuntu/pool/main/libv/libvpx/libvpx7_1.11.0-2ubuntu2.3_amd64.deb
        -this is the latest as at 12/16/2024 with a file date of 6/6/2024
5.  install libvpx7
        -in CLI/Terminal ensure you're in Downloads folder
        sudo apt install ./libvpx7_1.11.0-2ubuntu2.3_amd64.deb
        -this will add the needed virtualbox dependency
6.  if step 3 worked with no issues, then go to step 6a, otherwise do the following to install virtualbox 7.0
        sudo apt-get install virtualbox-7.0
        -all should install well
        -if this step fails then something else has happened and you will have to research the solution
6a.     -ensure the password you typed for the MOK enrollment is written down somewhere
        -reboot the PC
        -sign onto your encrypted disk if you have set that up
        -at the Perform MOK Management screen, do the following:
            -Enroll MOK - Continue
            -Enroll the key(s) - yes
            -enter the password you set either in step 3 or 6
            -select OK to reboot
        -your virtualbox cert should now be enrolled, if not, you'll have to research how to enroll the MOK
7.  add users to vboxusers group
        sudo usermod -aG vboxusers $USER
        -this adds the current user to the vboxusers group
        -if you want to add another user to this group then replace $USER with the user ID using the above command
        sudo newgrp vboxusers
        -to confirm you users are added to this group
        groups <username/ID>
        -this shows the group(s) the users belong to
8.  now download the virtualbox extension pack
        -in CLI/Terminal ensure you're in Downloads folder
        wget https://download.virtualbox.org/virtualbox/7.0.22/Oracle_VM_VirtualBox_Extension_Pack-7.0.22.vbox-extpack
9.  now install the extension pack
        sudo vboxmanage extpack install Oracle_VM_VirtualBox_Extension_Pack-7.0.22.vbox-extpack
10. in the Elementary OS GUI
        -click on Applications (top left)
        -open the Oracle VM Virtualbox application
        -click on File-Tools-Extension Pack Manager
        -confirm the Oracle VM Virtualbox Extension Pack is installed and the version is indeed 7.0.22
11. now download the guest additions iso for updating the drivers in your guest VMs
        wget https://download.virtualbox.org/virtualbox/7.0.22/VBoxGuestAdditions_7.0.22.iso
12. remove the following in /etc/apt/sources.list
        deb [arch=amd64 signed-by=/usr/share/keyrings/oracle-virtualbox-2016.gpg] https://download.virtualbox.org/virtualbox/debian jammy contrib
        -this has to be done otherwise any updates or new releases to virtualbox will have to be performed from step 1 again
13. [Optional] periodically backup your VMs to an alternate storage source if you're able to

At this point Virtualbox (7.0.22) is ready to create new VMs and/or ready to import/add existing VMs.

**Comment by davidmhewitt** [comment]
What's wrong with virtualbox from the ubuntu repositories? This is virtualbox 7 on elementary OS 8, so running `apt install virtualbox` would get you virtualbox without most of the instructions you provided.

If there is something wrong with the Ubuntu version, there are multiple issues with your instructions.
It can be useful to have generic Q&A resources like this for the community, but they should be safe and high quality. If you want to provide these instructions as a generic resource for other people, please explain the advantages of doing this instead of installing from the ubuntu repositories, and fix the below issues.

It would also be better to reformat your post as a question (How do I install virtualbox on elementary OS 8?) and then provide an "answer" to yourself to fit the Q&A format.

Why are you suggesting the Ubuntu "jammy" repositories when elementary OS 8 is based on Ubuntu "noble"?

You shouldn't edit sources.list directly and instead create a new entry in /etc/sources.list.d

Installing `linux-headers-$(uname -r)` is installing a specific version that won't get upgraded, which will probably cause issues with updates. Better to install `linux-headers-generic` which will automatically pull in the correct version.

It shouldn't be necessary to install dependencies separately. Doing this could cause security issues due to that package not being updated from the ubuntu repositories anymore and cause other issues on the system. This is probably necessary because the wrong distribution codename was used.

It's also bad advice to suggest disabling the virtualbox repository after adding it for security reasons. It's usually not good to have packages that aren't updatable. If it's breaking on updates, probably something was wrong with the installation method.
